### PR TITLE
Remove and removeMany changed to return the deleted data

### PR DESCRIPTION
**Pull request by IditYeger**
The changes include:
1. Two functions in the ctrl.ts file: remove() and removeMany(). They will return also the ResultData object. 
2. Mocking and testing for these changes.
3. MongoResult object's name changed to ResultData.

### CODE REVIEW

File: packages/db-controller/test/mocks/modelOptionsData.mock.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { ModelOptionsData, idType, idsType } from "../../src/wrappers";
+import { ResultData } from "../../src/wrappers/ResultData";
+import { ResultStatus } from "../../src/wrappers/ResultStatus";
+import { FilterData } from "../../src/filter-data";
+
+class ResponseMock {
+    status(something) { 
+      return responseStatusMock;
+    }
```
Comment (moshe-kabala) [inline packages/db-controller/test/mocks/modelOptionsData.mock.ts]: adding new  props to the mock class

data
status = 200 // defalut value

in status function 
status(s) {
this.status = status;
return this;
}

send(data) {
this.data = data;
return this
}

and in your test you can to check the
responseMock.data & responseMock.status

File: packages/db-controller/test/mocks/modelOptionsData.mock.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    send(something) {
+      return something;
+    }
+  }
+  
+  export const responseMock = new ResponseMock();
```
Comment (moshe-kabala) [inline packages/db-controller/test/mocks/modelOptionsData.mock.ts]: don't export as a single tone, simply create a new instance for each test

File: packages/db-controller/test/mocks/modelOptionsData.mock.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  
+  class ResponseStatusMock {
+    send(something) {
```
Comment (moshe-kabala) [inline packages/db-controller/test/mocks/modelOptionsData.mock.ts]: when you change the above class according to my comment this class will be a useless class.

File: packages/db-controller/test/mocks/modelOptionsData.mock.ts (hunk 1 above)
Comment (IditYeger) [reply packages/db-controller/test/mocks/modelOptionsData.mock.ts]: I fixed the code to something similar

File: packages/db-controller/test/mocks/modelOptionsData.mock.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import { ModelOptionsData, idType, idsType } from "../../src/wrappers";
+import { ResultData } from "../../src/wrappers/ResultData";
+import { ResultStatus } from "../../src/wrappers/ResultStatus";
+import { FilterData } from "../../src/filter-data";
+
+class ResponseMock {
+    status(something) {
```
[the remark this answers is not in the document]
Comment (IditYeger) [reply packages/db-controller/test/mocks/modelOptionsData.mock.ts]: Not sure I understand, I'll talk to you about it tomorrow

**Pull request merged**